.github/workflows/build-and-test.yml

```
@@ -53,3 +53,15 @@ jobs:
 
       - name: Run tests
         run: ./gradlew :lib:crypt4gh:test :lib:clearinghouse:test :lib:tsd-file-api-client:test :services:tsd-api-mock:test :services:mq-interceptor:test :services:localega-tsd-proxy:test :services:cega-mock:test :cli:lega-commander:test
+
+
+      - name: Run e2e tests
+        run: |
+          ./gradlew clean
+          ./gradlew start-docker-containers
+          sleep 20
```

```suggestion
              CONTAINERS=("tsd" "db" "mq" "proxy" "interceptor" 
                      "postgres" "ingest" "verify" "finalize" 
                      "mapper" "intercept" "doa" "cegamq" 
                      "cegaauth" "heartbeat-pub" "heartbeat-sub" 
                      "redis")

          # Wait for all containers to be running
          for container in "${CONTAINERS[@]}"; do
            echo "Waiting for $container to be running..."
            until [ "$(docker ps -q -f name="$container" -f status=running)" ]; do
              sleep 1
            done
            echo "$container is up and running."
          done
```
Maybe something like this?